### ISSUE TITLE
On Reaper 7x cycle actions in alternative keymaps do not work

### ISSUE THREAD

**Issue by Lo-lo78** [reaper-oss/sws#1803]
Sorry for the English! 🙂
I'm using the alternative keymaps that Reaper 7 offers.
Inserting cycle actions on main alt 1 or main alt 2 up to main alt 16 seems to have no effect.
Cycle actions only work on the main section.
Thanks for everything you do!

**Comment by cfillion** [comment]
What do you mean by "Inserting cycle actions on main alt 1" and "seems to have no effect"?

Cycle actions can not be inserted in the alt sections, the dropdown menu only in contains the primary one. Once created in Main you can assign them a different shortcut in the alt sections. That seems to work here.

**Comment by Lo-lo78** [comment]
Thanks Christian, I'm closing the issue.

**Comment by nofishonfriday** [comment]
https://forum.cockos.com/showthread.php?p=2764115#post2764115 ff

I can confirm this.
Made a cycle action only containing 'View: Move cursor right to grid division'.
Running this from Main section does move the cursor, but running it from one of the Main (alt) sections nothing happens here.

**Comment by cfillion** [comment]
Confirmed too, it seems to have broken in a recent(-ish?) REAPER update. It works in v7.0, for instance.

**Comment by nofishonfriday** [comment]
@cfillion Are you planning to look into it? Otherwise I/we could as first measure post in the 7.03(pre) thread to let the devs know that it broke cycle actions (in case they can fix it from Reaper side)?

**Comment by cfillion** [comment]
REAPER < 7.03:

```cpp
if(!hookcommand2(actual_section)) // no match
  hookcommand2(main_section);     // works
```

REAPER >= 7.03:

```cpp
if(!hookcommand2(actual_section)) // no match
  hookcommand2(actual_section);   // no match
```

The actual code is much more complicated (both calls are far apart, not even in the same function), which leads me to believe the old behavior might have been accidental.

All SWS main section actions that use `COMMAND_T::onAction` instead of `onCommand` are affected. In addition to Cycle Actions:

```
SWS/FNG: Apply selected groove (use curent settings from opened groove tool)
SWS/BR: Adjust playrate (MIDI CC only)
SWS/BR: Adjust playrate options...
SWS/BR: Play from mouse cursor position (perform until shortcut released)
SWS/BR: Play from mouse cursor position and solo track under mouse for the duration (perform until shortcut released)
SWS/BR: Play from mouse cursor position and solo item and track under mouse for the duration (perform until shortcut released)
SWS/BR: Play from edit cursor position (perform until shortcut released)
SWS/BR: Play from edit cursor position and solo track under mouse for the duration (perform until shortcut released)
SWS/BR: Play from edit cursor position and solo item and track under mouse for the duration (perform until shortcut released)
SWS/wol: Adjust selected envelope height (MIDI CC relative/mousewheel)
SWS/wol: Adjust selected envelope height, zoom center to middle arrange (MIDI CC relative/mousewheel)
SWS/wol: Adjust selected envelope height, zoom center to mouse cursor (MIDI CC relative/mousewheel)
SWS/wol: Adjust selected envelope or last touched track height (MIDI CC relative/mousewheel)
SWS/wol: Adjust selected envelope or last touched track height, zoom center to middle arrange (MIDI CC relative/mousewheel)
SWS/wol: Adjust selected envelope or last touched track height, zoom center to mouse cursor (MIDI CC relative/mousewheel)
SWS/wol: Adjust envelope or track height under mouse cursor (MIDI CC relative/mousewheel)
SWS/wol: Adjust envelope or track height under mouse cursor, zoom center to mouse cursor (MIDI CC relative/mousewheel)
SWS/S&M: Trigger preset for selected FX of selected track (MIDI/OSC only)
SWS/S&M: Select project (MIDI/OSC only)
SWS/S&M: Trigger preset for FX %d of selected track (MIDI/OSC only)
SWS/S&M: Live Config #%d - Apply config (MIDI/OSC only)
SWS/S&M: Live Config #%d - Preload config (MIDI/OSC only)
```

**Comment by Lo-lo78** [comment]
@cfillion 
Hi Chris I just tried with the latest SWS snapshot:
sws-2.14.0.1-Windows-x64-c_lanesettings-cf48dc8b.exe
I have done various tests and the cycle actions in the alternative keymaps from alt 1 to alt 16 do not work.
I created a cycle action that opens and closes all folders.
On the main section it works perfectly.
When I assign a shortcut to alt 1 it no longer works.
Looks like maybe we should report to Cockos?
I don't know if SWS can fix this problem.
Thank you!
